third_party/indexed-db-as-promised/src/sync-promise.js

```diff
@@ -0,0 +1,194 @@
+export default class SyncPromise {
+  constructor(resolver) {
+    if (!isFunction(resolver)) {
+      throw new TypeError('Must pass resolver function');
+    }
+
+    this._state = PendingPromise;
+    this._value = [];
+
+    doResolve(
+      this,
+      adopter(this, FulfilledPromise),
+      adopter(this, RejectedPromise),
+      { then: resolver }
+    );
+  }
+
+  then(onFulfilled, onRejected) {
+    onFulfilled = isFunction(onFulfilled) ? onFulfilled : void 0;
+    onRejected = isFunction(onRejected) ? onRejected : void 0;
+
+    return this._state(
+      this._value,
+      onFulfilled,
+      onRejected
+    );
+  }
+
+  catch(onRejected) {
+    return this.then(void 0, onRejected);
+  }
+
+  static resolve(value) {
+    const Constructor = this;
```

Comment:
Actually this will not work in closure compiler code I believe. Only reason not to take a hard ref on the class is to allow for inheritance, which I think would be bad here.